Source/MVC6/Boilerplate.AspNetCore.TagHelpers/Twitter/Cards/TwitterCardApp.cs

```diff
@@ -105,5 +101,33 @@ public override void ToString(StringBuilder stringBuilder)
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:url:googleplay", this.GooglePlayCustomUrlScheme);
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:country", this.Country);
         }
+
+        /// <summary>
+        /// Checks that this instance is valid and throws exceptions if not valid.
+        /// </summary>
+        protected override void Validate()
+        {
+            base.Validate();
+
+            if (string.IsNullOrEmpty(this.SiteUsername))
+            {
+                throw new ArgumentNullException(nameof(this.SiteUsername));
+            }
+
+            if (string.IsNullOrEmpty(this.IPhone))
```

Comment:
I can't believe that iPhone, iPad and GooglePlay are all required. It's assumed that all apps exist!